website_sale_product_reference_displayed/views/website_sale_views.xml

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2020 Tecnativa - David Vidal
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <template id="products_item" name="Display product reference" inherit_id="website_sale.products_item">
+        <!-- Product title in list -->
+        <xpath expr="//a[@t-field='product.name']" position="attributes">
+            <attribute name="class" add="hidden" separator=" "/>
+        </xpath>
+        <xpath expr="//a[@t-field='product.name']" position="after">
+            <a itemprop="name" t-att-href="keep('/shop/product/%s' % slug(product), page=(pager['page']['num'] if pager['page']['num']&gt;1 else None))" t-att-content="product.display_name" t-field="product.display_name"/>
+        </xpath>
+    </template>
+
+    <template id="product" name="Display product reference" inherit_id="website_sale.product">
+        <xpath expr="//t[@t-set='additional_title']" positions="after">
+            <t t-set="additional_title" t-value="product.display_name" />
+        </xpath>
+        <!-- Breadcrumb -->
+        <xpath expr="//span[@t-field='product.name']" position="attributes">
+            <attribute name="class" add="hidden" separator=" "/>
+        </xpath>
+        <xpath expr="//span[@t-field='product.name']" position="after">
+            <span t-field="product.display_name"/>
+        </xpath>
+        <!-- Product Title -->
+        <xpath expr="//h1[@t-field='product.name']" position="attributes">
+            <attribute name="class" add="hidden" separator=" "/>
+        </xpath>
+        <xpath expr="//h1[@t-field='product.name']" position="after">
+            <h1 itemprop="name" t-field="product.display_name">Product Name</h1>
```

Comment:
This print "Product Name" if 'display_name' is empty?

Reply:
Yeah, I just copied the original from Odoo, but it's useless indeed